### PR TITLE
CLOUDTRUST-2225: Add missing credentials APIs

### PR DESCRIPTION
**Pull request by lagess**
retrieved from commented APIs in KC8

### CODE REVIEW

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    @PUT
+    @Consumes(javax.ws.rs.core.MediaType.TEXT_PLAIN)
+    @Path("{credentialId}/label")
+    public void setLabel(final @PathParam("credentialId") String credentialId, String userLabel) {
+        auth.require(AccountRoles.MANAGE_ACCOUNT);
+        session.userCredentialManager().updateCredentialLabel(realm, user, credentialId, userLabel);
+    }
+
+    /**
+     * Move a credential to a position behind another credential
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Move a credential to the first position ?

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            event.error(org.keycloak.events.Errors.INVALID_USER_CREDENTIALS);
+            return ErrorResponse.error(Messages.INVALID_PASSWORD_EXISTING, Response.Status.BAD_REQUEST);
+        }
+        
+        if (update.getNewPassword() == null) {
+            return ErrorResponse.error(Messages.INVALID_PASSWORD_EXISTING, Response.Status.BAD_REQUEST);
+        }
+        
+        String confirmation = update.getConfirmation();
+        if ((confirmation != null) && !update.getNewPassword().equals(confirmation)) {
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: String.equals(...) supports case when parameter is null so we don't need to check for `confirmation!=null`

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java (hunk 1 above)
Comment (lagess) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: fixed

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java (hunk 2 above)
Comment (lagess) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: As this is copy-pasted KC code, I'll let it as is

Comment (fperot74) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Thanks for the information: this is something I'll have to submit in my PR with cleaning of their code ;-)

Comment (lagess) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: :-)

**Pull request merged**